src/lib.rs

```diff
@@ -69,6 +69,19 @@ pub mod contract {
         pub ticks: Vec<Tick>,
     }
 
+    #[derive(scale::Decode, Default, scale::Encode, Clone, Debug)]
+    #[cfg_attr(
+        feature = "std",
+        derive(scale_info::TypeInfo, ink::storage::traits::StorageLayout,)
+    )]
+    pub struct SwapParams {
+        pool_key: PoolKey,
+        x_to_y: bool,
+        amount: TokenAmount,
+        by_amount_in: bool,
+        sqrt_price_limit: SqrtPrice,
```

Comment:
There should be one limit. Set all limits to the maximum, and finally, check slippage based on the number of expected last hop tokens